File: Source/Easy_Roman_To_Integer.cs
using System;
using System.Collections.Generic;

public class Solution
{
	public int Roman_To_Integer(string s)
	{
		int integer = 0;
		Dictionary<char, int> numerals = new Dictionary<char, int>();

		numerals.Add('I', 1);
		numerals.Add('V', 5);
		numerals.Add('X', 10);
		numerals.Add('L', 50);
		numerals.Add('C', 100);
		numerals.Add('D', 500);
		numerals.Add('M', 1000);

		for(int i = 0; i < s.Length; i++)
		{
			if(i + 1 >= s.Length)
			{
				integer += numerals[s[i]];
			}

			if(i + 1 < s.Length)
			{
				if(numerals[s[i]] >= numerals[s[i + 1]]) integer += numerals[s[i]];
				if(numerals[s[i]] < numerals[s[i + 1]]) {integer += numerals[s[i + 1]] - numerals[s[i]]; i++;}
			}
		}

		return integer;
	}
}

class LeetCode
{
	public static void Main()
	{
		string s1 = "IV";
		string s2 = "IX";
		string s3 = "III";
		string s4 = "LVIII";
		string s5 = "MMCXVI";
		string s6 = "MCMXCIV";
		Solution solution = new Solution();
		Console.WriteLine(solution.Roman_To_Integer(s6));
	}
}

File: Source/Easy_Roman_To_Integer.c
#include <stdio.h>

int Numerals(char c)
{
	switch(c)
	{
		case 'I' : return 1;
		case 'V' : return 5;
		case 'X' : return 10;
		case 'L' : return 50;
		case 'C' : return 100;
		case 'D' : return 500;
		case 'M' : return 1000;
	}

	return 0;
}

int Roman_To_Integer(char* s)
{
	int integer = 0;

	for(int i = 0; i < strlen(s); i++)
	{
		if(i + 1 >= strlen(s))
		{
			integer += Numerals(s[i]);
		}

		if(i + 1 < strlen(s))
		{
			if(Numerals(s[i]) >= Numerals(s[i + 1])) integer += Numerals(s[i]);
			if(Numerals(s[i]) < Numerals(s[i + 1])) {integer += Numerals(s[i + 1]) - Numerals(s[i]); i++;}
		}
	}

	return integer;
}

char* s1 = "IV";
char* s2 = "IX";
char* s3 = "III";
char* s4 = "LVIII";
char* s5 = "MMCXVI";
char* s6 = "MCMXCIV";

int main()
{
	printf("%d", Roman_To_Integer(s6));
}